File: xchanger/src/main/java/com/finsim/xchanger/dossiers/model/Dossier.java
package com.finsim.xchanger.dossiers.model;

import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "dossiers")
public class Dossier {
    @Id
    private String id;

    private String name;
    private String surname;
    private String birth;

    private String abi;
    private String externalId;
    @Indexed(unique = true)
    private String iban;

    private List<DossierStock> stocks;
    private DossierValue value;

    public DossierDto toDto() {
        return new DossierDto(
            this.id,
            this.name,
            this.surname,
            this.birth,
            this.abi,
            this.externalId,
            this.iban,
            this.stocks,
            this.value
        );
    }
}

File: mainframe-lib/xchanger/model/dossier.go
package model

type Dossier struct {
	Id         string         `json:"id"`
	Name       string         `json:"name"`
	Surname    string         `json:"surname"`
	Birth      string         `json:"birth"`
	Abi        string         `json:"abi"`
	ExternalId string         `json:"externalId"`
	IBAN       string         `json:"iban"`
	Stocks     []DossierStock `json:"stocks,omitempty"`
	Value      DossierValue   `json:"value"`
}

type DossierStock struct {
	Isin      string `json:"isin"`
	Total     int32  `json:"total"`
	Available int32  `json:"available"`
}

type DossierValue struct {
	Value     Price  `json:"value"`
	Timestamp string `json:"timestamp"`
}

type InsertDossierInput struct {
	Name       string `json:"name"`
	Surname    string `json:"surname"`
	Birth      string `json:"birth"`
	ExternalId string `json:"externalId"`
	IBAN       string `json:"iban"`
}
